src/common/csv_reader/csv_reader.cpp

```diff
@@ -324,5 +325,70 @@ void CSVReader::openFile(const string& fName) {
     }
 }
 
+CopyDescription::CopyDescription(const string& filePath, CSVReaderConfig readerConfig)
+    : filePath{filePath}, readerConfig{nullptr}, fileType{FileType::CSV} {
+    setFileType(filePath);
+    if (fileType == FileType::CSV) {
+        this->readerConfig = make_unique<CSVReaderConfig>(readerConfig);
+    }
+}
+
+CopyDescription::CopyDescription(const CopyDescription& copyDescription)
+    : filePath{copyDescription.filePath}, readerConfig{nullptr}, fileType{
+                                                                     copyDescription.fileType} {
+    if (fileType == FileType::CSV) {
```

Comment:
Check for this null or not condition directly on the readerConfig field. So something like:
```
if (copyDescription.readerConfig) {
  this->readerConfig = make_unique<CSVReaderConfig>(*copyDescription.readerConfig);
}
```